### PR TITLE
Fixing changes from aakash' PR

### PR DESCRIPTION
**Pull request by NicholasTanYY**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -519,6 +527,95 @@ public static void deleteAllTasksOnDate (TaskManager taskManager, LocalDate spec
...
+
+### Managing Free Time
+
+<a name="free-times-manager-method"></a>
+
+### Free Times Manager Method
+
+The `freeTimesManager` method facilitates the identification and display of free time slots on a specific date. It is a crucial feature for users to efficiently plan their schedules by identifying available time intervals.
+
+#### Method Signature
```
Comment (kyhjonathan) [inline docs/DeveloperGuide.md:540]: LGTM!

File: text-ui-test/changeExpected.sh [hunk 2]
```diff
@@ -1,9 +1,9 @@
 #!/usr/bin/env bash
 
 # reset saved file from previous run
-if [ -e "save/tasks.txt" ]
+if [ -e "save/clintonData.txt" ]
```
Comment (clarencepohh) [inline text-ui-test/changeExpected.sh:4]: Good catch on refactored save file name!

File: src/test/java/data/TaskManagerTest.java [hunk 3]
```diff
@@ -363,12 +365,17 @@ void updateDeadlineDescriptionAndByDateTime_validInput_updatesTask() throws Task
...
-        assertEquals(updatedByTime, taskManager.getTasksForDate(date).get(0).getByTime());
-        assertEquals(updatedByDate, taskManager.getTasksForDate(date).get(0).getByDate());
+        assertEquals(updatedTaskDescription, taskManager.getTasksForDate(LocalDate.parse(updatedByDate,
+                DateTimeFormatter.ofPattern("dd/MM/yyyy"))).get(0).getName());
+        assertEquals(updatedByTime, taskManager.getTasksForDate(LocalDate.parse(updatedByDate,
+                DateTimeFormatter.ofPattern("dd/MM/yyyy"))).get(0).getByTime());
+        assertEquals(updatedByDate, taskManager.getTasksForDate(LocalDate.parse(updatedByDate,
+                DateTimeFormatter.ofPattern("dd/MM/yyyy"))).get(0).getByDate());
+
+        deleteAllTasksOnDate(taskManager, LocalDate.now().plusDays(2));
```
Comment (clarencepohh) [inline src/test/java/data/TaskManagerTest.java:378]: Good way to stop our text-ui-tests from failing!

**Review by clarencepohh — COMMENTED** [review]
Good job fixing!

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -519,6 +527,95 @@ public static void deleteAllTasksOnDate (TaskManager taskManager, LocalDate spec
...
+3. Iterates through the events and calculates free time slots between events.
+4. Adds the calculated free time slots to the list.
+5. Returns the list of free time slots.
+
+<a name="printing-free-time-slots"></a>
+
+### Printing Free Time Slots
+
+The `printFreeTimeSlots` method prints the identified free time slots for a specific date.
+
```
Comment (Aak242) [inline docs/DeveloperGuide.md:600]: Thanks for streamlining the changes made on my PR!

File: docs/diagrams/class/Exceptions.puml [hunk 5]
```diff
@@ -2,48 +2,55 @@
...
 
 class SetPriorityException extends TaskManagerException {
-    + TASK_INDEX_OUT_OF_RANGE_FOR_DAY_WITH_TASKS_MESSAGE : String
-    + TASK_INDEX_WITH_NO_TASKS_MESSAGE : String
-    + INVALID_PRIORITY_MESSAGE : String
-    + SetPriorityException(errorMessage : String)
-    + checkIfTaskIndexIsValidForPriority(dayTasks : List<Task>, taskIndex : int) : void
-    + checkIfPriorityIsValid(priority : String) : void
+
+{static} + TASK_INDEX_OUT_OF_RANGE_FOR_DAY_WITH_TASKS_MESSAGE : String
```
Comment (Mahesh1772) [inline docs/diagrams/class/Exceptions.puml:37]: Good catch of static being used!

**Review by Aak242 — COMMENTED** [review]
LGTM!

File: docs/diagrams/sequence/UpdateTaskSequenceDiagram.puml [hunk 6]
```diff
@@ -0,0 +1,85 @@
+@startuml
```
Comment (Mahesh1772) [inline docs/diagrams/sequence/UpdateTaskSequenceDiagram.puml:1]: Good job on the UML diagram!

File: docs/images/Update-Deadline-Example.png
Comment (Mahesh1772) [inline docs/images/Update-Deadline-Example.png:1]: Thanks for changing it!

**Review by Mahesh1772 — APPROVED** [review]
LGTM!

**Pull request merged**